Build failure: pkgsCross.mingwW64.windows.dlfcn

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

-

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
```

### Additional context

* tracking issue: #445447

### System metadata

x86_64-linux

### Notify maintainers

@marius851000

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/windows/dlfcn/default.nix
+++ b/pkgs/os-specific/windows/dlfcn/default.nix
@@ -8,13 +8,13 @@
 
 stdenv.mkDerivation rec {
   pname = "dlfcn";
-  version = "1.3.1";
+  version = "1.4.2";
 
   src = fetchFromGitHub {
     owner = "dlfcn-win32";
     repo = "dlfcn-win32";
     tag = "v${version}";
-    hash = "sha256-ljVTMBiGp8TPufrQcK4zQtcVH1To4zcfBAbUOb+v910=";
+    hash = "sha256-kLY8vvHTT02gENPlVvyDyR0ULC8NA+E/P6mWtU6MbBY=";
   };
 
   nativeBuildInputs = [ cmake ];

--- a/pkgs/os-specific/windows/dlfcn/default.nix
+++ b/pkgs/os-specific/windows/dlfcn/default.nix
@@ -12,17 +12,17 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "dlfcn-win32";
     repo = "dlfcn-win32";
-    rev = "v${version}";
-    sha256 = "sha256-ljVTMBiGp8TPufrQcK4zQtcVH1To4zcfBAbUOb+v910=";
+    tag = "v${version}";
+    hash = "sha256-ljVTMBiGp8TPufrQcK4zQtcVH1To4zcfBAbUOb+v910=";
   };
 
   nativeBuildInputs = [ cmake ];
 
-  meta = with lib; {
+  meta = {
     homepage = "https://github.com/dlfcn-win32/dlfcn-win32";
     description = "Set of functions that allows runtime dynamic library loading";
-    license = licenses.mit;
-    platforms = platforms.windows;
-    maintainers = with maintainers; [ marius851000 ];
+    license = lib.licenses.mit;
+    platforms = lib.platforms.windows;
+    maintainers = with lib.maintainers; [ marius851000 ];
   };
 }

--- a/pkgs/os-specific/windows/dlfcn/default.nix
+++ b/pkgs/os-specific/windows/dlfcn/default.nix
@@ -3,6 +3,7 @@
   lib,
   fetchFromGitHub,
   cmake,
+  nix-update-script,
 }:
 
 stdenv.mkDerivation rec {
@@ -17,6 +18,8 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ cmake ];
+
+  passthru.updateScript = nix-update-script { };
 
   meta = {
     homepage = "https://github.com/dlfcn-win32/dlfcn-win32";

